File: app/src/main/java/com/jinjoo/pouch/activity/Activity.java
package com.jinjoo.pouch.activity;

import android.app.ProgressDialog;

import androidx.appcompat.app.AppCompatActivity;

public class Activity extends AppCompatActivity {
    ProgressDialog pd;
    public void showPd(){
        if(pd == null){
            pd = new ProgressDialog(this);
            pd.setCancelable(true);
            pd.setMessage("잠시만 기다려주세요...");
        }
        pd.show();
    }

    public void stopPd(){
        if(pd != null && pd.isShowing()){
            pd.dismiss();
        }
    }
}

File: app/src/main/kotlin/com/jinjoo/pouch/activity/Activity.kt
package com.jinjoo.pouch.activity

import android.app.ProgressDialog
import androidx.appcompat.app.AppCompatActivity

open class Activity : AppCompatActivity() {
    val pd: ProgressDialog by lazy { ProgressDialog(this) }
    fun showPd() {
        with(pd){
            setCancelable(true)
            setMessage("잠시만 기다려주세요...")
            show()
        }
    }

    fun stopPd() {

        if (pd.isShowing) {
            pd.dismiss()
        }
    }
}